lib/workers/repository/process/lookup/index.ts

```diff
@@ -379,7 +384,7 @@ function getFromVersion(
   config: LookupUpdateConfig,
   rangeStrategy: string,
   allVersions: string[]
-): string {
+): string | null {
```

Comment:
Is this change required?

Reply:
No, but it’s an unambiguous improvement since the function was already capable of returning `null` and it’s helpful for the type to document that fact.

At the moment `strictNullChecks` is turned off in `tsConfig.json` for compatibility with the parts of renovate that are still written in JavaScript, but as you convert more of the code to TypeScript you probably want to work towards turning this option on to get better static type checking.

Reply:
Also I wanted to make it clear from the type of the function that

```typescript
getFromVersion(config, rangeStrategy, nonDeprecatedVersions) ||
      getFromVersion(config, rangeStrategy, allVersions)
```

works because the first function call returns `null` if there are no matching non-deprecated versions.